coderd/externalauth/externalauth.go

```diff
@@ -95,17 +95,33 @@ func (c *Config) GenerateTokenExtra(token *oauth2.Token) (pqtype.NullRawMessage,
 	}, nil
 }
 
+// InvalidReason is intentionally typed and not an error. An invalid token is not an error,
+// but a state. A boolean true/false omits context and information that could be helpful in debugging.
+// So a string is returned with some additional information.
+type InvalidReason string
+
+// Valid returns true if there is no reason to be invalid.
+func (r InvalidReason) Valid() bool {
+	return r == ""
```

Comment:
An `InvalidReason` cannot be `Valid` with its name. It would be nil, but not valid.